### PR TITLE
fix: update replication configuration format

### PR DESCRIPTION
**Pull request by rahulreddy**
This commit updates the replication configuration format that gets
written to config.json. This format updates the properties from
name/endpoint to site/servers.

### CODE REVIEW

**Comment by rahulreddy** [comment]
@ironman-machine 
SCALITY_ARSENAL_BRANCH=fix/updateReplicationConfigFormat

**Comment by rahulreddy** [comment]
@ironman-machine 
SCALITY_ARSENAL_BRANCH=fix/updateReplicationConfigFormat
try

File: lib/Config.js [hunk 1]
```diff
@@ -237,24 +237,26 @@ class Config {
                 assert.strictEqual(typeof replicationEndpoint, 'object',
                     'bad config: `replicationEndpoints` property must be an ' +
                     'array of objects');
-                const { name, endpoint } = replicationEndpoint;
-                assert.notStrictEqual(name, undefined, 'bad config: each ' +
+                const { site, servers } = replicationEndpoint;
+                assert.notStrictEqual(site, undefined, 'bad config: each ' +
                     'object of `replicationEndpoints` array must have a ' +
                     '`name` property');
```
Comment (bennettbuchanan) [inline lib/Config.js]: `site` property.

File: lib/Config.js [hunk 2]
```diff
@@ -237,24 +237,26 @@ class Config {
...
-                    '`name` property of object in `replicationEndpoints` ' +
+                assert.strictEqual(typeof site, 'string', 'bad config: ' +
+                    '`site` property of object in `replicationEndpoints` ' +
                     'must be a string');
-                assert.notStrictEqual(name, '', 'bad config: `name` property ' +
+                assert.notStrictEqual(site, '', 'bad config: `site` property ' +
                     "of object in `replicationEndpoints` must not be ''");
-                assert.notStrictEqual(endpoint, undefined, 'bad config: each ' +
+                assert.notStrictEqual(servers, undefined, 'bad config: each ' +
                     'object of `replicationEndpoints` array must have an ' +
```
Comment (bennettbuchanan) [inline lib/Config.js]: ~an~ a

File: lib/Config.js [hunk 3]
```diff
@@ -237,24 +237,26 @@ class Config {
...
-                assert.strictEqual(typeof endpoint, 'string', 'bad config: ' +
-                    '`endpoint` property of object in `replicationEndpoints` ' +
+                    '`servers` property');
+                assert(servers instanceof Array, 'bad config: ' +
+                    '`servers` property of object in `replicationEndpoints` ' +
+                    'must be an array');
+                servers.forEach(item => {
+                    assert(typeof item === 'string' && item !== '',
+                    'bad config: each item of `replicationEndpoints:servers` ' +
                     'must be a string');
```
Comment (bennettbuchanan) [inline lib/Config.js]: must be a non-empty string.

File: lib/Config.js (hunk 1 above)
Comment (philipyoo) [inline lib/Config.js]: Minor change here to "site"?

Comment (rahulreddy) [inline lib/Config.js]: Yes, good catch!

**Pull request merged**